### PR TITLE
EKA-612 changing means of bundle parsing

### PR DESCRIPTION
**Pull request by angshu**
Also fixed issue of Consent request status not changing to GRANTED on notification

### CODE REVIEW

File: src/main/java/in/org/projecteka/hiu/consent/GrantedConsentTask.java [hunk 1]
```diff
@@ -51,10 +52,11 @@ public GrantedConsentTask(ConsentRepository consentRepository,
...
                     var cmSuffix = getCmSuffix(consentRequest.getPatient().getId());
-                    return Flux.fromIterable(consentNotification.getConsentArtefacts())
-                            .flatMap(reference -> perform(reference, consentNotification.getConsentRequestId(),
-                                    cmSuffix))
-                            .then();
+                    return consentRepository.updateConsentRequestStatus(
+                            ConsentStatus.GRANTED, consentNotification.getConsentRequestId())
+                            .then(Flux.fromIterable(consentNotification.getConsentArtefacts())
+                                    .flatMap(reference -> perform(reference, consentNotification.getConsentRequestId(), cmSuffix))
+                                    .then());
```
Comment (dheeraj-p) [inline src/main/java/in/org/projecteka/hiu/consent/GrantedConsentTask.java:59]: Why is this needed?

**Review by dheeraj-p — APPROVED** [review]

File: src/main/java/in/org/projecteka/hiu/consent/GrantedConsentTask.java (hunk 1 above)
Comment (angshu) [reply src/main/java/in/org/projecteka/hiu/consent/GrantedConsentTask.java:59]: we were not updating the consent_request status in case of GRANT.
